### PR TITLE
GIX-1353: pollAccounts in participation flow

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

If user opens Participate in sale modal and there are no accounts, request and retry until we get them.

# Changes

* Call `pollAccounts` on mount of ParticipateSwapModal.svelte
* Fix `pollAcconuts` to use 2 seconds in milliseconds.

# Tests

* Fix broken tests.
* Add test in ParticipateSwapModal to check that accounts are requested if not present.
* Fix tests after new functionality in the modal.

### CODE REVIEW

**Comment by lmuntaner** [comment]
New UX:

I mocked failing the first 4 calls to `queryAccount`:
https://user-images.githubusercontent.com/4550653/221904191-2826ea52-9670-406c-92fb-6fede3bc178f.mov

**Comment by lmuntaner** [comment]
@dskloetd please review

File: frontend/src/tests/lib/components/project-detail/ParticipateButton.spec.ts [hunk 1]
```diff
@@ -93,6 +98,7 @@ describe("ParticipateButton", () => {
       await waitFor(() =>
         expect(getByTestId("transaction-step-1")).toBeInTheDocument()
       );
+      accountsStore.reset();
```
Comment (dskloetd) [inline frontend/src/tests/lib/components/project-detail/ParticipateButton.spec.ts:101]: Remove or put this in beforeEach()

**Review by dskloetd — APPROVED** [review]

File: frontend/src/tests/lib/modals/sns/ParticipateSwapModal.spec.ts [hunk 2]
```diff
@@ -94,88 +102,114 @@ describe("ParticipateSwapModal", () => {
...
+        .mockResolvedValue(mockAccountDetails);
+      await renderSwapModal();
+
+      await waitFor(() => expect(queryAccountSpy).toBeCalledTimes(1));
+      expect(queryAccountBalanceSpy).toBeCalledTimes(1);
+      expect(queryAccountBalanceSpy).toBeCalledWith({
+        accountIdentifier: mockMainAccount.identifier,
+        certified: true,
+        identity: mockIdentity,
+      });
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply frontend/src/tests/lib/modals/sns/ParticipateSwapModal.spec.ts]: done

File: frontend/src/tests/lib/components/project-detail/ParticipateButton.spec.ts (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/tests/lib/components/project-detail/ParticipateButton.spec.ts]: done

**Pull request merged**